pkg/resources/object_renaming_lists_and_sets.go

```diff
@@ -0,0 +1,518 @@
+		oldOrderedListMapped := mapObjectRenamingDatabaseOrderedListItemFromValue(oldOrderedList)
+		newOrderedListMapped := mapObjectRenamingDatabaseOrderedListItemFromValue(newOrderedList)
+
+		itemsToAdd, itemsToRemove := ListDiff(oldOrderedListMapped, newOrderedListMapped)
+		for _, removedItem := range itemsToRemove {
+			ObjectRenamingDatabaseInstance.OrderedList = slices.DeleteFunc(ObjectRenamingDatabaseInstance.OrderedList, func(item objectRenamingDatabaseOrderedListItem) bool {
+				return item == removedItem
+			})
+		}
+
+		for _, addedItem := range itemsToAdd {
+			addedItem.Order = ""
+			ObjectRenamingDatabaseInstance.OrderedList = append(ObjectRenamingDatabaseInstance.OrderedList, addedItem)
+		}
+
+		// The implementation is not complete due to mentioned issues with computed order in DiffSuppressFunc
+	}
+
+	if d.HasChange("manually_ordered_list") {
+		_, newManuallyOrderedList := d.GetChange("manually_ordered_list")
+		hasNewItems := len(newManuallyOrderedList.([]any)) > len(ObjectRenamingDatabaseInstance.ManuallyOrderedList)
+
+		// Copy the value from external source and compare with the new state to create the final state that will be saved
+		finalState := ObjectRenamingDatabaseInstance.ManuallyOrderedList
+
+		for index, item := range ObjectRenamingDatabaseInstance.ManuallyOrderedList {
+			order := index + 1
+			newValue, err := collections.FindFirst(newManuallyOrderedList.([]any), func(item any) bool {
+				return item.(map[string]any)["order"].(int) == order
+			})
+
+			// Items were removed
+			if errors.Is(err, collections.ErrObjectNotFound) {
+				finalState = finalState[:index]
+				break
+			}
+
+			newValueName := (*newValue).(map[string]any)["name"].(string)
+			// Rename item (handle item change)
+			if newValueName != item.Name {
+				log.Printf("[DEBUG] Renaming item at order %d from name %s to %s", order, item.Name, newValueName)
+				ObjectRenamingDatabaseInstance.ManuallyOrderedList[index].Name = newValueName
+			}
+		}
+
+		// Add remaining items
+		if hasNewItems {
+			maxOrder := len(finalState)
+			newMaxOrder := len(newManuallyOrderedList.([]any))
+			for order := maxOrder + 1; order <= newMaxOrder; order++ {
+				item, err := collections.FindFirst(newManuallyOrderedList.([]any), func(item any) bool {
+					return item.(map[string]any)["order"].(int) == order
+				})
+				if err != nil {
+					return diag.FromErr(fmt.Errorf("couldn't find item to add with order %d", order))
+				}
+
+				finalState = append(finalState, ObjectRenamingDatabaseManuallyOrderedListItem{
+					Name: (*item).(map[string]any)["name"].(string),
+				})
+			}
+		}
+
+		ObjectRenamingDatabaseInstance.ManuallyOrderedList = finalState
+	}
+
+	return ReadObjectRenamingListsAndSets(ctx, d, meta)
+}
+
+func ReadObjectRenamingListsAndSets(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	list := collections.Map(ObjectRenamingDatabaseInstance.List, func(t objectRenamingDatabaseListItem) map[string]any {
```

Comment:
Currently, It would be only used in the orderedList read. I don't dislike the idea, I was thinking about adding it, but maybe next pr or wait for the other use case for it (it's a 2-3 lines of improvement right now).